lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Sign.hs

```diff
@@ -226,16 +234,45 @@ estimateKeyWitnessCount utxo tx =
         in
             nonInputWits <> inputWits
   where
-    CardanoApi.Tx txbody@(CardanoApi.TxBody txbodycontent) _keyWits
+    CardanoApi.Tx (CardanoApi.TxBody txbodycontent) _keyWits
         = toCardanoApiTx tx
 
-    scripts = case txbody of
-        CardanoApi.ShelleyTxBody _ _ shelleyBodyScripts _ _ _ ->
-            shelleyBodyScripts
-        CardanoApi.ByronTxBody {} ->
-            error "estimateKeyWitnessCount: ByronTxBody"
+    timelockScriptVkCount :: Natural
+    timelockScriptVkCount = sum $ Map.elems $ Map.unionWith
+        (\_est spec -> spec)
+        upperBoundEstimatedTimelockVkCounts
+        specifiedTimelockVkCounts
+      where
+        specifiedTimelockVkCounts :: Map (ScriptHash StandardCrypto) Natural
+        specifiedTimelockVkCounts = Map.fromList $ mapMaybe resolve
+            $ F.toList scriptsNeeded
+          where
+            resolve
+                :: (ScriptHash StandardCrypto)
+                -> Maybe (ScriptHash StandardCrypto, Natural)
+            resolve h = (h,) <$> Map.lookup h
+                (getSpecifiedTimelockScriptVkCounts timelockVkCounts)
+
+        upperBoundEstimatedTimelockVkCounts
+            :: Map (ScriptHash StandardCrypto) Natural
+        upperBoundEstimatedTimelockVkCounts = Map.mapMaybe
+            (fmap (estimateMaxWitnessRequiredPerInput . toCAScript)
+                . toTimelockScript)
```

Comment:
Suggestion: make it easier to see where the first parenthesis is terminated:

```suggestion
            ( fmap (estimateMaxWitnessRequiredPerInput . toCAScript)
            . toTimelockScript
            )
```

Reply:
IMO the light usage of lisp-style parenthesis looks fine